controllers/clustercache/cluster_accessor.go

```diff
@@ -495,3 +497,80 @@ func (ca *clusterAccessor) unlock(ctx context.Context) {
 	ca.lockedStateLock.Unlock()
 	log.V(10).Info("Removed lock for ClusterAccessor")
 }
+
+// ScopedWatcher is a scoped-down interface from Controller that only has the Watch func.
```

Comment:
nit: I would prefer if we move everything below here to cluster_cache.go

cluster_cache.go contains most (all?) of the public parts of the ClusterCache. cluster_accessor.go is just an implementation detail